golang/cosmos/x/lien/keeper/keeper.go

```diff
@@ -100,39 +103,69 @@ func (lk keeperImpl) IterateLiens(ctx sdk.Context, cb func(addr sdk.AccAddress,
 	}
 }
 
-// AccountState represents the abstract state of an account.
-// See ../spec/01_concepts.md for details.
-type AccountState struct {
-	Total     sdk.Coins `json:"total"`
-	Bonded    sdk.Coins `json:"bonded"`
-	Unbonding sdk.Coins `json:"unbonding"`
-	Locked    sdk.Coins `json:"locked"`
-	Liened    sdk.Coins `json:"liened"`
-}
+// UpdateLien changes the liened amount of a single denomination in the given account.
+// Either the old or new amount of the denomination can be zero.
+// Liens can always be decreased, but to increase a lien, the new total amount must
+// be vested and either bonded (for the staking token) or in the bank (for other tokens).
```

Comment:
We may want to revisit: why can't liens be taken against staking token in the bank?